src/algebra/algebra/subalgebra.lean

```diff
@@ -598,6 +598,19 @@ instance : unique (subalgebra R R) :=
   end
   .. algebra.subalgebra.inhabited }
 
+/-- Two subalgebras that are equal are also equivalent as algebras. -/
+@[simps apply]
+def equiv_of_eq (S T : subalgebra R A) (h : S = T) : S ≃ₐ[R] T :=
+{ to_fun := λ x, ⟨x, h ▸ x.2⟩,
+  inv_fun := λ x, ⟨x, h.symm ▸ x.2⟩,
+  map_mul' := λ _ _, rfl,
+  commutes' := λ _, rfl,
+  .. linear_equiv.of_eq _ _ (congr_arg to_submodule h) }
+
+@[simp] lemma equiv_of_eq_symm (S T : subalgebra R A) (h : S = T) :
+  (equiv_of_eq S T h).symm = equiv_of_eq T S h.symm :=
+rfl
+
```

Comment:
I know we don't have it for `linear_equiv.of_eq`, but a lemma about `equiv_of_eq S T rfl = alg_equiv.refl` would probably be a good addition. One about `(equiv_of_eq S T h).trans (equiv_of_eq T R h2) = equiv_of_eq S R (h.trans h2)` is less useful, but likely worth having for completeness